packages/use-i18n/src/usei18n.tsx

```diff
@@ -23,53 +23,54 @@ import type { ReactParamsObject, ScopedTranslateFn, TranslateFn } from './types'
 
 const LOCALE_ITEM_STORAGE = 'locale'
 
+type SupportedLocalesType<LocalSupportedType extends string> = (
+  locale: string,
+) => locale is LocalSupportedType
+
 type TranslationsByLocales = Record<string, BaseLocale>
-type RequiredGenericContext<Locale extends BaseLocale> =
-  keyof Locale extends never
-    ? Omit<Context<Locale>, 't' | 'namespaceTranslation'> & {
-        t: (str: 'You must pass a generic argument to useI18n()') => void
-        namespaceTranslation: (
-          str: 'You must pass a generic argument to useI18n()',
-        ) => void
-      }
-    : Context<Locale>
+type RequiredGenericContext<
+  LocaleParam extends BaseLocale,
+  LocalSupportedType extends string,
+> = keyof string extends never
```

Comment:
```suggestion
> = keyof LocaleParam extends never
```

Keyof string extends never will be always false ? We try to check the generic here

Reply:
indeed ! good catch